### DISCUSSION TITLE
Getting all headings after parsing to html

### DISCUSSION THREAD

**Discussion by 4e576rt8uh9ij9okp** [markedjs/marked#2461] [Q&A] [answered]
Hello,
I'm making a static site generator using just MD files and ETA templating.
I want to create a Table of Content and I would need to access the parsed heading of the file. I assume that you store it somewhere because you put the heading names into the headers id and I would need to access all of the headings, like a list or object.

Is there a callback or something where I can access it?

**Comment by UziTech** [comment, marked as answer]
You can use [`walkTokens`](https://marked.js.org/using_pro#walk-tokens) to run a function for each token and collect the heading tokens to create a TOC.

**Reply by 4e576rt8uh9ij9okp** [reply]
This helped, I did this here:
```js
// Marked config
function walkTokens(token){
    if(token.type === "heading"){
        marked.headings.push(token.text)
    }
}

marked.use({ walkTokens })
```

I have also another function with `marked.headings = []` inside to always reset it for each file being parsed.
